src/Types/Tonal.elm

```diff
@@ -0,0 +1,96 @@
+module Types.Tonal exposing (Accidental(..), Letter(..), Pitch(..), Note(..), PitchNotation, chromaticScale, letterSemitoneOffset, accidentalSemitoneOffset)
+
+import Array exposing (Array)
+import Types.Octave exposing (..)
+
+
+type Accidental
+    = DoubleFlat
+    | Flat
+    | Natural
+    | Sharp
+    | DoubleSharp
+
+
+type Letter
+    = C
+    | D
+    | E
+    | F
+    | G
+    | A
+    | B
+
+
+type Note
+    = Note Letter Accidental
+
+
+type Pitch
```

Comment:
What is reason for the extra pitch.
```elm
type Pitch = Pitch Note Octave
```
vs.
```elm
type Pitch = Note Octave
```

The former seems easier to read when the type is used. Is there any other reason for it?

Reply:
type Pitch = Note Octave

This doesn't work. Because `Note` would be the constructor in this case.

There are several options.

1. Let `Pitch` be an alias for a tuple:

    type alias Pitch = ( Note, Octave)

2. Alias for a record:

    type alias Pitch
        = { note : Note,
           , octave : Octave
           }

3. Opaque type with record:

    type Pitch
        = Pitch
               { note : Note
               , octave : Octave
               }

4. Or as in our case:

     type Pitch
            = Pitch Note Octave     

I'm not 100% sure what is best, still experimenting. So far I like the current approach.